src/runtime/entitlements-manager.js

```diff
@@ -356,75 +411,159 @@ export class EntitlementsManager {
       .callbacks()
       .triggerEntitlementsResponse(Promise.resolve(entitlements));
 
-    // Show a toast if needed.
-    this.maybeShowToast_(entitlements);
+    const entitlement = entitlements.getEntitlementForThis();
+    if (!entitlement) {
+      return;
+    }
+
+    this.maybeShowToast_(entitlement);
   }
 
   /**
-   * @param {!Entitlements} entitlements
+   * @param {!Entitlement} entitlement
    * @return {!Promise}
    * @private
    */
-  maybeShowToast_(entitlements) {
-    const entitlement = entitlements.getEntitlementForThis();
-    if (!entitlement) {
-      return Promise.resolve();
-    }
+  maybeShowToast_(entitlement) {
     // Check if storage bit is set. It's only set by the `Entitlements.ack`
     // method.
     return this.storage_.get(TOAST_STORAGE_KEY).then((value) => {
-      if (value == '1') {
-        // Already shown;
+      const toastWasShown = value === '1';
+      if (toastWasShown) {
         return;
       }
-      if (entitlement) {
-        this.showToast_(entitlement);
-      }
+
+      // Show toast.
+      const source = entitlement.source || 'google';
+      return new Toast(
+        this.deps_,
+        feUrl('/toastiframe'),
+        feArgs({
+          'publicationId': this.publicationId_,
+          'source': source,
+        })
+      ).open();
     });
   }
 
   /**
-   * @param {!Entitlement} entitlement
+   * @param {!Entitlements} entitlements
    * @private
    */
-  showToast_(entitlement) {
-    const source = entitlement.source || 'google';
-    return new Toast(
-      this.deps_,
-      feUrl('/toastiframe'),
-      feArgs({
-        'publicationId': this.publicationId_,
-        'source': source,
-      })
-    ).open();
+  ack_(entitlements) {
+    if (entitlements.getEntitlementForThis()) {
+      this.setToastShown(true);
+    }
   }
 
   /**
    * @param {!Entitlements} entitlements
    * @private
    */
-  ack_(entitlements) {
-    if (entitlements.getEntitlementForThis()) {
-      this.setToastShown(true);
+  consume_(entitlements) {
+    if (entitlements.enablesThisWithGoogleMetering()) {
+      // TODO: Open Metering Prompt.
+      const activityIframeView_ = new ActivityIframeView(
+        this.win_,
+        this.deps_.activities(),
+        feUrl('/contributionsiframe'),
```

Comment:
contributionsiframe for consume + google metering?
or is this a placeholder for something?
I don't get it

Reply:
This is just a placeholder. I will add a comment mentioning this